testdb/delete.js

```diff
@@ -0,0 +1,87 @@
+const { entity, field } = require('gotu')
+const Repository = require('../src/repository')
+const db = require('./db')
+const config = require('./config')
+const assert = require('assert')
+
+describe('Delete an Entity', () => {
+
+    const table = 'test_repository'
+    const schema = 'gotu2knex_testdb'
+
+    before(async () => {
+        const sql = `
+        DROP SCHEMA IF EXISTS ${schema} CASCADE;
+        CREATE SCHEMA ${schema};
+        DROP TABLE IF EXISTS ${schema}.${table} CASCADE; 
+        CREATE TABLE ${schema}.${table} (
+            id INT,
+            string_test TEXT,
+            boolean_test BOOL,
+            PRIMARY KEY (id)
+        );
+        INSERT INTO ${schema}.${table} values (1, 'created', true)`
+        await db.query(sql)
+    })
+
+    after(async () => {
+        const sql = `
+        DROP SCHEMA IF EXISTS ${schema} CASCADE;
+        `
+        await db.query(sql)
+    })
+
+    const givenAnRepositoryClass = (options) => {
+        return class ItemRepositoryBase extends Repository {
+            constructor() {
+                super(options)
+            }
+        }
+    }
+
+    describe('Simple entity', () => {
+
+        const givenAnEntity = () => {
+            return entity('A entity', {
+                id: field(Number),
+                stringTest: field(String),
+                booleanTest: field(Boolean)
+            })
+        }
+
+        const givenAnModifiedEntity = () => {
+            const anEntity = givenAnEntity()
+            const anEntityInstance = new anEntity()
+            anEntityInstance.id = 1
+            anEntityInstance.stringTest = "test"
+            anEntityInstance.booleanTest = true
+            return anEntityInstance
+        }
+
+        it('should delete an existing item', async () => {
+
+            //given
+            const anEntity = givenAnEntity()
+            const ItemRepository = givenAnRepositoryClass({
+                entity: anEntity,
+                table,
+                schema,
+                ids: ['id'],
+                dbConfig: config
+            })
+            const aModifiedInstance = givenAnModifiedEntity()
+
+            const injection = {}
+            const itemRepo = new ItemRepository(injection)
+
+            //when
+            aModifiedInstance.stringTest = "delete"
+            const ret = await itemRepo.delete(aModifiedInstance)
+
+            //then
+            const retDB = await db.query(`SELECT string_test FROM ${schema}.${table} WHERE id = ${aModifiedInstance.id}`)
+            assert.deepStrictEqual(ret, true)
+            assert.deepStrictEqual(retDB.rows[0].string_test, "delete")
```

Comment:
If we'll to delete the registry from DB, how it will return any value and compare with "delete"?

Reply:
Done!